source/Weather.cpp

```diff
@@ -70,12 +71,13 @@ const Point &Weather::Origin() const
 
 
 // Create any environmental effects and decrease the lifetime of this weather.
-void Weather::Step(vector<Visual> &visuals)
+void Weather::Step(vector<Visual> &visuals, const Ship *flagship)
 {
 	// Environmental effects are created by choosing a random angle and distance from
 	// their origin, then creating the effect there.
 	double minRange = hazard->MinRange();
-	double maxRange = hazard->MaxRange();
+	// If it is systemwide, only draw around the flagship.
+	double maxRange = !hazard->SystemWide() ? hazard->MaxRange() : 5000.;
```

Comment:
Reads better without negation:

```suggestion
	double maxRange = hazard->SystemWide() ? 5000. : hazard->MaxRange();
```
(Also note that there is no guarantee that 5000 is large enough for the viewport.)

Reply:
I had 10 000 before but it seemed overkill considering that is until the system edge, meaning quite big
Also it seemed a bit overkill to go and check the monitor used, and the zoom, I mostly didn't do it because I'm not sure how much distance means etc